aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -187,6 +204,13 @@ impl GasAlgebra for StandardGasAlgebra {
         &mut self,
         abstract_amount: impl GasExpression<VMGasParameters, Unit = InternalGasUnit>,
     ) -> PartialVMResult<()> {
+        if self.block_synchronization_kill_switch.interrupt_requested() {
```

Comment:
Do you need it for IO? I think we only use it in `charge_change_set` which is called at the end of user session.

Reply:
I wanted to have it everywhere to be oblivious to how these APIs are used, maybe we should even more it to charge() later.